patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
```

Comment:
Oh thanks, I copied most of this content here from StackOverflow. I opened those two (and indeed both libraries use the pattern now) but I can gladly link to other libraries I had nothing to do with that happen to use the pattern. It's pretty common and the reason I linked to it from the StackOverflow post is that it includes the debate which I thought was interesting.